Automattic/wp-calypso
# Previews: Improvements to design, accessibility and ability to use it inline

This PR brings many improvements to all preview modals (site, posts, pages, …) and also introduces a way to embed the preview inlined and not just in the modal window. We will use this base to create a new view with site preview and a menu item in sidebar "View Site" like this:

<img width="1440" alt="screen shot 2017-06-02 at 00 54 35" src="https://cloud.githubusercontent.com/assets/156676/26727641/d6880f64-47a7-11e7-9278-8e1fc4755bd3.png">

New stuff (dotcom-specific for now, Jetpack will come next):

- previewed page shows before all images are loaded - less waiting and much faster feedback for user
- previewed page is now focused after being displayed, so you can use arrow keys or other means of keyboard interaction with the page without clicking the page first
- we handle ESC key being pressed while the previewed page is focused (previously ESC was noop)
- "Edit" links that WordPress puts under all pages/posts etc don't open a new tab with Calypso but it rather closes the preview and navigates you to the appropriate page without any reload

This all needs few updates on the backend. Available in D5791-code.

New stuff for any site:

- device switcher + SEO got combined into one smaller component, leaving us space for further updates in the Toolbar that we plan
- "SEO" got renamed to "Search & Social", which explains the functionality much better

- device switch will get icons later, once #14617 is merged

This is just first PR that will allow us to create much more stuff view previews soon.

#### Test instructions

1. boot branch
2. patch your sandbox with D5791-code
3. pick a dotcom site that you'll test with and point it also to your sandbox
4. make sure all previews work correctly (site preview, post preview from list, post preview from editor, …)
5. try new features listed above
6. you can try /preview/yoursite.wordpress.com but that might be rather broken for now. Don't worry too much, it's not linked anywhere but allows us to work on the feature in more people before making it active for everyone

Fixes #2001
